src/libAtomVM/nifs.c

```diff
@@ -4229,6 +4235,7 @@ static term nif_code_load_binary(Context *ctx, int argc, term argv[])
         }
         memcpy(allocated_data, term_binary_data(binary), bin_size);
         data = allocated_data;
+        free(allocated_data);
```

Comment:
This introduces a bug I think, `data` here is a dangling pointer.

Reply:
I was unsure about this one, I'm surprised that
test_code_load_binary.erl passed, maybe that test is not thorough enough to catch this.

Reply:
I see why this is a bad change, and now understand why it is still passed the test.

I dropped this commit.